docs/manual/file_mode.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: default
+title: rpm.org - File Mode
+---
+# File Mode
+
+RPM supports a variety of compression methods through external libraries for
+its internal I/O operations.  For ease of use, the [RPMIO
+API](https://ftp.osuosl.org/pub/rpm/api/4.17.0/group__rpmio.html) offers
+special `fopen(3)/fdopen(3)`-like functions that accept an extended *mode
+string* which allows for specifying a compression method alongside the usual
+`r` or `w` letters.  RPM will then perform (de)compression transparently for
+the caller.
+
+While mostly used internally, a mode string can also be exposed through
+configuration options or macros where appropriate.  Currently, only payload
+write mode can be configured this way, see the [Payload](format.html#payload)
+section for details.
+
+## Format
+
+The format of an RPMIO mode string is as follows:
+
+```
+<mode>[flags].<type>
+```
+
+The `mode` parameter is mandatory and contains a sequence of letters as
+described in `fopen(3)`.  For payload configuration, only `w` is allowed.
+
+The `type` parameter is also mandatory and specifies the compression type
+(*compressor*) to be used.  Currently, the following compressors are supported:
+
+| Type		| Description					    |
+|---------------|---------------------------------------------------|
+|`gzdio`	| gzip ([flags](#gzip-flags))			    |
+|`bzdio`	| bzip2 ([flags](#bzip2-flags))			    |
+|`xzdio`	| lzma ([flags](#xz-lzma-flags))		    |
```

Comment:
Just call it XZ in the description too, the real state of affairs here is just too confusing for the uninitiated :smile:

Reply:
...especially as the examples below talk about "xz"

Reply:
Yeah, not gonna lie, I was always confused by this xz/lzma thing, too :sweat_smile:  OK, will do that, thanks!